src/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -102,7 +102,7 @@ internal static SafeSslHandle AllocateSslContext(SslProtocols protocols, SafeX50
                     if (sslAuthenticationOptions.IsServer)
                     {
                         byte[] protos = Interop.Ssl.ConvertAlpnProtocolListToByteArray(sslAuthenticationOptions.ApplicationProtocols);
-                        sslAuthenticationOptions.AlpnProtocolsHandle = GCHandle.Alloc(protos);
+                        sslAuthenticationOptions.AlpnProtocolsHandle = GCHandle.Alloc(protos, GCHandleType.Pinned);
```

Comment:
@Priya91, is AlpnProtocolsHandle only used on Linux?  It appears to also be in but not used in the other builds.

Reply:
I'm wondering if it should actually be stored on the SafeSslHandle, in which case its ReleaseHandle would take care of cleaning it up.

Reply:
Yeah I did that, it broke, because the tests are disposing before doing the authenticate.... we need some sort of lock here or proper cancellation support

Reply:
Yeah the protocols handle is used only on Linux, that's how the os apis expect the protocols to be passed.

Reply:
> that's how the os apis expect the protocols to be passed.

Ok, then we shouldn't be paying for it in other builds and should ensure that state and associated logic is only in Linux-specific files.

Reply:
Agreed, on this however, due to the current way SslStream tears down, it leaves a pretty big window if someone disposes SslStream for OpenSSL to access this pinned buffer (which isn't pinned anymore). Now this is bad user code, but its worse that it leads to memory corruption. I would say leave it for now, and tidy it up either through reworking the tear down, or removing that pin. Either of which will need a PR of their own later. 

Windows calling the `if(Handle.IsAllocated)` for now on object destruction seems a small price to pay to avoid this.